comment_scan/replier.rb

```diff
@@ -80,68 +82,104 @@ def random_response
     end
 
     def contains_cat(message)
-        ["cat", "kitty", "kitties", "kitten", "meow", "purr", "feline"].any? { |cat_name| message.downcase.include? cat_name }
+        ["cat", "kitty", "kitties", "kitten", "kitteh", "meow", "purr", "feline"].any? { |cat_name| message.downcase.include? cat_name }
     end
 
-    def tp(msg_id, parent_id, room_id, *args)
-        comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
-        return if comment.nil?
-
-        comment.tps ||= 0
-        comment.tps += 1
-        comment.save
+    def contains_dog(message)
+        ["dog", "pup", "woof", "bark", "bow-wow", "best friend"].any? { |dog_name| message.downcase.include? dog_name }
+    end
 
-        @chatter.say "Marked this comment as caught correctly (tp). Currently marked #{comment.tps.to_i}tps/#{comment.fps.to_i}fps. *beep boop* My human overlords won't let me flag that, so you'll have to do it yourself.", room_id
+    def animals_on?(room_id)
+      room_id == @chatter.HQroom || Room.find_by(room_id: room_id).animals 
     end
 
-    def fp(msg_id, parent_id, room_id, *args)
+
+    def record_feedback(feedback_id, parent_id, chat_user, room_id)
         comment = MessageCollection::ALL_ROOMS.comment_for(parent_id.to_i)
-        return if comment.nil?
+        return false if comment.nil?
 
+        comment.tps ||= 0
         comment.fps ||= 0
-        comment.fps += 1
-        comment.save
-        @chatter.say "Marked this comment as caught incorrectly (fp). Currently marked #{comment.tps.to_i}tps/#{comment.fps.to_i}fps", room_id
+        comment.rude ||= 0
+
+        existing_feedback = Feedback.find_by(comment_id: comment.id, chat_user_id: chat_user.id)
+
+        if existing_feedback.nil? #Create new feedback
+            Feedback.create(comment_id: comment.id, chat_user_id: chat_user.id, feedback_type_id: feedback_id, room_id: room_id)
+        elsif existing_feedback.feedback_type_id #If one exists, then undo it
```

Comment:
Why undo it instead of simply updating the feedback type?

Reply:
So I do actually just update the feedback type down on line 134. However, what needs to be undone is the tp/fp/rude count on the comment table. This implementation continues to support that in addition to adding feedbacks. If we got rid of those tp/fp/rude columns altogether and solely relied on the new feedback table (so getting those counts would be a feedback query on the comment id), then this whole thing could be a lot simpler. Maybe sometime in the future we could have a script go back and convert any tp/fp/rude numbers into feedback table rows with a -1 user id?

Reply:
I totally agree with that, we shouldn't have information duplicated like that. Want to throw that in an issue?

Reply:
Ooh good call. Just added that.